include/psa/crypto_values.h

```diff
@@ -2021,6 +2021,18 @@
 #define PSA_ALG_TLS12_PSK_TO_MS_GET_HASH(hkdf_alg)                         \
     (PSA_ALG_CATEGORY_HASH | ((hkdf_alg) & PSA_ALG_HASH_MASK))
 
+/* The TLS 1.2 ECJPAKE-to-PMS KDF. It takes the shared secret K (an EC point
+ * in case of EC J-PAKE) and calculates SHA256(K.X) that the rest of TLS 1.2
+ * will use to derive the session secret, as defined by step 2 of
+ * https://datatracker.ietf.org/doc/html/draft-cragie-tls-ecjpake-01#section-8.7.
+ * Uses PSA_ALG_SHA_256.
+ * This function takes a single input:
+ * #PSA_KEY_DERIVATION_INPUT_SECRET is the shared secret K from EC J-PAKE.
+ * The only supported curve is secp256r1 (the 256-bit curve in
+ * #PSA_ECC_FAMILY_SECP_R1), so the input must be exactly 65 bytes.
+ */
+#define PSA_ALG_TLS12_ECJPAKE_TO_PMS            ((psa_algorithm_t)0x08000600)
```

Comment:
If there is a possibility that this algorithm might in future use a different hash algorithm, then it might be useful to define this one with the Hash algorithm type value for SHA-256 (`0x09`) in the low 8 bits of the identifier. i.e. `0x08000609`.

If there is no chance that this might happen, then the selected value is acceptable for the PSA Crypto API.

Reply:
I really don't see this algorithm being updated in the future, as it's for TLS 1.2 which is now officially the past (as of 4 years ago). Also, if there's a version of ECJPAKE for TLS 1.3 some day, it will likely not use its own ad-hoc KDF but the more standard HKDF-Extract instead, as is customary in TLS 1.3.

That said, however confident I may feel, predictions are always hard, especially about the future, and picking `0x08000609` doesn't seem to hurt in any way, so perhaps we should just go with that?

Reply:
Even if no new algorithm is ever added, using a uniform encoding of the hash might help to simplify the implementation.

Reply:
By the way, there's a precedent: EdDSA is defined as using one specific hash for each curve, and the algorithm values do encode that hash in the lower 8 bits.